addons/web/static/src/js/views/graph/graph_renderer.js

```diff
@@ -80,6 +82,16 @@ return AbstractRenderer.extend({
      */
     updateState: function (state, params) {
         this.isComparison = !!state.comparisonData;
+        if (params.context !== undefined) {
+            var timeRangeMenuData = params.context.timeRangeMenuData;
+            if (timeRangeMenuData) {
+                this.timeRangeDescription = timeRangeMenuData.timeRangeDescription;
+                this.comparisonTimeRangeDescription = timeRangeMenuData.comparisonTimeRangeDescription;
+            } else {
+                this.timeRangeDescription = undefined;
+                this.comparisonTimeRangeDescription = undefined;
+            }
+        }
```

Comment:
i'd move all this to the model directly, and store the description info into the state (alongside other timerange info)